### PR TITLE
Add math.log2 support

### PR DESCRIPTION
**Pull request by guilhermeleobas**
Partially fixes https://github.com/numba/numba/issues/9414

### CODE REVIEW

File: docs/upcoming_changes/9416.new_feature.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Support math.log2
+=================
```
Comment (stuartarchibald) [inline docs/upcoming_changes/9416.new_feature.rst:2]:
```suggestion
Add support for ``math.log2``.
==============================
```

File: numba/cpython/mathimpl.py [hunk 2]
```diff
@@ -176,6 +176,7 @@ def float_impl(context, builder, sig, args):
 exp_impl = unary_math_intr(math.exp, 'llvm.exp')
 log_impl = unary_math_intr(math.log, 'llvm.log')
 log10_impl = unary_math_intr(math.log10, 'llvm.log10')
+log2_impl = unary_math_intr(math.log2, 'llvm.log2')
```
Comment (stuartarchibald) [inline numba/cpython/mathimpl.py:179]: Technically this implementation should raise `ValueError('math domain error')` for input values that are zero or negative, however, what is present is no different to e.g. `log10` above which would suffer from the same issue.

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch. Please could you add that this is supported to the docs here:
https://github.com/numba/numba/blob/3edb458c69daeaaffbfc938145c367eda1cf043f/docs/source/reference/pysupported.rst?plain=1#L1076
one suggestion RE release note else looks good.

File: numba/cpython/mathimpl.py (hunk 2 above)
Comment (guilhermeleobas) [reply numba/cpython/mathimpl.py:179]: I can tackle this in a follow-up PR if needed.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Pull request merged**